easy-ui-react/src/Notification/Notification.tsx

```diff
@@ -0,0 +1,344 @@
+ * showSuccessToast(content: ToastProps): void;
+ * showWarningToast(content: ToastProps): void;
+ * showErrorToast(content: ToastProps): void;
+ * showPromotionalAlert(content: AlertProps): void;
+ * showNeutralAlert(content: AlertProps): void;
+ * showSuccessAlert(content: AlertProps): void;
+ * showWarningAlert(content: AlertProps): void;
+ * showErrorAlert(content: AlertProps): void;
+ * 
+ * type ToastProps = {
+ *   // Notification message
+ *   message: ReactNode;
+ *   // Notification will render with left aligned status icon
+ *   hasIcon?: boolean;
+ * }
+ * 
+ * type AlertProps = {
+ *   // Notification message
+ *   message: ReactNode;
+ *   // Notification will render with left aligned status icon
+ *   hasIcon?: boolean;
+ *   // Callback function when dismissing alerts
+ *   onDismiss?: () => void;
+ * }
+ * ```
+ * @example
+ * _Success:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*  const onToast = () => {
+*    notification.showSuccessToast({ message: "message" });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showSuccessAlert({ message: "message" });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _Without status icons:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*  const onToast = () => {
+*    notification.showPromotionalToast({ message: "message", hasIcon: false });
+*  };
+*
+*  const onAlert = () => {
+*    notification.showWarningAlert({ message: "message", hasIcon: false });
+* };
+*
+*  return (
+*    <>
+*      <Button onPress={onToast}>Show toast</Button>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+* @example
+* _With onDismiss callback, only applicable for alerts:_
+```tsx
+* import { useNotification } from "@easypost/easy-ui/Notification";
+* import { Button } from "@easypost/easy-ui/Button";
+*
+* function Component() {
+*  const { notification } = useNotification();
+*
+*
+*  const onAlert = () => {
+*    notification.showNeutralAlert({
+*      message: "message",
+*      onDismiss: () => alert("i am called immediately before the alert closes"),
+*    });
+*  };
+*
+*  return (
+*    <>
+*      <Button onPress={onAlert}>Show alert</Button>
+*    </>
+*  );
+*}
+*```
+ */
+export function useNotification() {
+  const notification = useContext(NotificationContext);
+  if (!notification) {
+    throw new Error(
+      "useNotification() must be used within a NotificationProvider, which is bundled with EasyUIProvider",
+    );
+  }
+  return { notification: notification };
```

Comment:
**question:** what other properties are returned on `notification` other than the `show*Toast` and `show*Alert` methods given that it also extends `ToastState<NotificationProps>` as well? follow up is do we want to expose that to the consumer?

Reply:
my initial thought was that we do want to include some of the properties included by `ToastState` [here ](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/toast/src/useToastState.ts#LL48C33-L48C33) but after thinking about it a bit more, i think it has the potential to become unnecessarily confusing for consumers having the `show<Status><Type>` methods alongside a generic `add`. i was thinking maybe theres a use case for needing programmatic dismissal and having the ability to pause toasts, but again, i think it could become too confusing. i updated the PR so that we're only exposing the `show<Status><Type>` methods.

Reply:
got ya. i agree regarding `add`. i think it would be confusing being in addition to our `show` methods. however i also agree that programmatic dismissal may come in handy.

a specific use case i'm wondering about: in apps, many times it's desired to close all toasts on page navigation. i'm wondering how that could be accomplished with what we have exposed?

Reply:
thats a good point, i went and added the ability to programmatically dismiss notifications via `closeActiveNotification()`

on navigation on our setup, it should be happening as is without needing to do it programmatically

https://github.com/EasyPost/easy-ui/assets/44451175/219993fe-d0da-4c5e-ae05-d0bb38f936b1